hack/verify-imports.sh

```diff
@@ -43,4 +43,4 @@ if [[ ! -x "$importverifier" ]]; then
   exit 1
 fi
 
-"${importverifier}" "k8s.io/" "${KUBE_ROOT}/staging/publishing/import-restrictions.yaml"
+"${importverifier}" "k8s.io/" "${KUBE_ROOT}/staging/publishing/import-restrictions.yaml" --disallowed-import "github.com/pkg/errors" --ignored-path "cmd/kubeadm/app"
```

Comment:
would this restriction and functionality make more sense in the hack/unwanted-dependencies.json file?

Reply:
Well, I didn't know that this file exists!
I agree, it's a little bit odd.

But As I can see from `dependencyverifier`, this will not satisfy what we are looking for.
`dependencyverifier` will execute `go mod graph` and check the output, but this will also check for cross`vendored` dependencies, I know we can handle it!

But Actually, What we need is to check the import in go files! and not the `go.mod`,
And we need also to ignore some sub-trees. 

For instance, in our case, we need to check if there are imports of those `unwanted dependencies` more specific `github.com/pkg/errors` in the k8s packages including the staging. but we should ignore the check in some specific sub-path, `cmd/kubeadm/app` in the case of `errors`.


so, what do you suggest?
```
- We can move this logic to `dependencyverifier`
- We can extend `dependencyverifier` in order to check imports in go files as well ?
- ... anything else?
```

Reply:
if we want package-level enforcement preventing use of this, that seems more like the existing `.import-restrictions` files sprinkled throughout the codebase which can match and forbid dependencies on specific packages

Reply:
I took a look into it, let me check again. what I can do.